src/Processors/Transforms/FilterTransform.cpp

```diff
@@ -271,5 +258,47 @@ void FilterTransform::doTransform(Chunk & chunk)
     chunk.setColumns(std::move(columns), num_filtered_rows);
 }
 
+void FilterTransform::writeIntoQueryConditionCache(MarkRangesInfoPtr mark_info)
+{
+    if (!query_condition_cache)
+        return;
+
+    if (!mark_info)
+    {
+        if (!matching_mark_info)
+            return;
+
+        query_condition_cache->write(
+            matching_mark_info->table_uuid,
```

Comment:
maybe pass `matching_mark_info` as a whole instead of every it's data member

Reply:
https://github.com/zhongyuankai/ClickHouse/blob/dd155c5c5cc4f2adbe25366ffb96dd771cff744a/src/Storages/MergeTree/MergeTreeSelectProcessor.cpp#L177C1-L178C1
The `write` method will also be called here, and it will be more general to pass it separately.